client/src/templates/Challenges/components/multiple-choice-questions.tsx

```diff
@@ -7,74 +7,106 @@ import ChallengeHeading from './challenge-heading';
-  ) => void;
+  questions: Question[];
+  selectedOptions: (number | null)[];
+  handleOptionChange: (questionIndex: number, answerIndex: number) => void;
+  submittedMcqAnswers: (number | null)[];
+  showFeedback: boolean;
 };
 
+function removeParagraphTags(text: string): string {
+  return text.replace(/^<p>|<\/p>$/g, '');
+}
+
 function MultipleChoiceQuestions({
-  questions: { text, answers },
-  selectedOption,
-  isWrongAnswer,
-  handleOptionChange
+  questions,
+  selectedOptions,
+  handleOptionChange,
+  submittedMcqAnswers,
+  showFeedback
 }: MultipleChoiceQuestionsProps): JSX.Element {
   const { t } = useTranslation();
 
-  const feedback =
-    selectedOption !== null ? answers[selectedOption].feedback : undefined;
-
   return (
     <>
       <ChallengeHeading heading={t('learn.question')} />
-      <PrismFormatted className={'line-numbers'} text={text} />
-      <div className='video-quiz-options'>
-        {answers.map(({ answer }, index) => (
-          <label
-            className='video-quiz-option-label'
-            key={index}
-            htmlFor={`mc-question-${index}`}
-          >
-            <input
-              name='quiz'
-              checked={selectedOption === index}
-              className='sr-only'
-              onChange={handleOptionChange}
-              type='radio'
-              value={index}
-              id={`mc-question-${index}`}
-            />{' '}
-            <span className='video-quiz-input-visible'>
-              {selectedOption === index ? (
-                <span className='video-quiz-selected-input' />
-              ) : null}
-            </span>
-            <PrismFormatted
-              className={'video-quiz-option'}
-              text={answer.replace(/^<p>|<\/p>$/g, '')}
-              useSpan
-              noAria
-            />
-          </label>
-        ))}
-      </div>
-      {isWrongAnswer && (
-        <>
-          <Spacer size='medium' />
-          <div className='text-center'>
-            {feedback ? (
-              <PrismFormatted
-                className={'multiple-choice-feedback'}
-                text={feedback}
-              />
-            ) : (
-              t('learn.wrong-answer')
-            )}
+      {questions.map((question, questionIndex) => (
+        <div key={questionIndex}>
+          <PrismFormatted className={'line-numbers'} text={question.text} />
+          <div className='video-quiz-options'>
+            {question.answers.map(({ answer }, answerIndex) => {
+              const isSubmittedAnswer =
+                submittedMcqAnswers[questionIndex] === answerIndex;
+              const feedback =
+                questions[questionIndex].answers[answerIndex].feedback;
+              const isCorrect =
+                submittedMcqAnswers[questionIndex] ===
+                // -1 because the solution is 1-indexed
+                questions[questionIndex].solution - 1;
+
+              return (
+                <>
```

Comment:
```suggestion
                <React.Fragment key={answerIndex}>
```
The fragment needs the key (since it's the wrapping element)